core/dive.h

```diff
@@ -608,7 +608,10 @@ static inline unsigned int number_of_computers(struct dive *dive)
 
 static inline struct divecomputer *get_dive_dc(struct dive *dive, int nr)
 {
-	struct divecomputer *dc = &dive->dc;
+	struct divecomputer *dc;
+	if (!dive)
+		return NULL;
+	dc = &dive->dc;
```

Comment:
you could have left that one line declaration / value assignment, just put the if-clause first. But this is fine, too.